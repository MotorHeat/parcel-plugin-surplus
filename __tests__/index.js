/* eslint-env jest */
const inputPath = `${__dirname}/samples/input.js`

describe('#initialization', () => {
  const surplusPlugin = require('../index')
  const Bundler = require('parcel-bundler')
  let bundler = new Bundler(inputPath)
  surplusPlugin(bundler)
  const jsAsset = 'parcel-plugin-surplus/lib/js-asset'

  it('surplus plugin should be a function', () =>
    expect(typeof surplusPlugin).toBe('function'))

  it('should define SurplusAsset as the js asset', () =>
    expect(bundler.parser.extensions['.js'].includes(jsAsset)).toBeTruthy())
  it('should define SurplusAsset as the jsx asset', () =>
    expect(bundler.parser.extensions['.jsx'].includes(jsAsset)).toBeTruthy())
  it('should define SurplusAsset as the es6 asset', () =>
    expect(bundler.parser.extensions['.es6'].includes(jsAsset)).toBeTruthy())
  it('should define SurplusAsset as the jsm asset', () =>
    expect(bundler.parser.extensions['.jsm'].includes(jsAsset)).toBeTruthy())
  it('should define SurplusAsset as the mjs asset', () =>
    expect(bundler.parser.extensions['.mjs'].includes(jsAsset)).toBeTruthy())
})

describe('#transform', () => {
  const Bundler = require('parcel-bundler')
  const surplusPlugin = require('../index')
  let bundler = new Bundler(inputPath, {
    watch: false
  })
  surplusPlugin(bundler)

  it('should transform the jsx in input.js', async () => {
    expect.assertions(1)
    const result = await bundler.bundle()
    expect(result.entryAsset.generated.js.trim()).toMatchSnapshot()
  })
})
