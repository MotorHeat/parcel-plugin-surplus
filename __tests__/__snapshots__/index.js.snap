// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#transform should transform the jsx in input.js 1`] = `
"\\"use strict\\";

var Surplus = _interopRequireWildcard(require(\\"surplus\\"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var view = function () {
  var __, __h21;

  __ = Surplus.createElement(\\"div\\", null, null);
  __h21 = Surplus.createElement(\\"h2\\", null, __);
  __h21.textContent = \\"Surplus\\";
  return __;
}();

document.body.appendChild(view);"
`;
